controllers/articles.controller.js

```diff
@@ -0,0 +1,12 @@
+const { selectArticleById } = require("../models/articles.model");
+
+exports.getArticleById = (req, res, next) => {
+  return selectArticleById(req)
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({ status: 404, msg: "Not found" });
```

Comment:
We should be destructuring the rows and returning them from the model rather than in here. The promise.reject would therefore also be there. The controller should only be for invoking the model and sending the response. Please move this logic into the model.

Reply:
as john said, the check of the result array length,, and promise reject if it is 0, should be in the model